src/vs/workbench/services/hover/browser/hoverWidget.ts

```diff
@@ -443,6 +436,25 @@ export class HoverWidget extends Widget {
 		}
 	}
 
+	private adjustHoverMaxHeight(target: TargetRect): void {
+		let maxHeight = window.innerHeight / 2;
+
+		// When force position is enabled, restrict max height
+		if (this._forcePosition) {
+			const padding = (this._hoverPointer ? Constants.PointerSize : 0) + Constants.HoverBorderWidth;
+			if (this._hoverPosition === HoverPosition.ABOVE) {
+				maxHeight = Math.min(maxHeight, target.top - padding);
+			} else if (this._hoverPosition === HoverPosition.BELOW) {
+				maxHeight = Math.min(maxHeight, window.innerHeight - target.bottom - padding);
+			}
+		}
+
+		// Make sure not to accidentally enlarge the hover when setting a maxHeight for it
+		maxHeight = Math.min(maxHeight, this._hover.containerDomNode.clientHeight);
```

Comment:
I'm thinking this might be redundant but it was a worry expressed in https://github.com/microsoft/vscode/pull/163879#discussion_r999836240